packages/e2e-test-utils/src/templates.js

```diff
@@ -47,7 +47,7 @@ export async function deleteAllTemplates( type ) {
 			);
 		}
 
-		if ( ! response.deleted ) {
+		if ( ! response?.deleted ) {
```

Comment:
When the response is not defined? Does it come from a failing request that we `catch` above?

Reply:
That was my guess as well. We can also `continue` the loop when a REST API request throws an error.

Reply:
> When the response is not defined? Does it come from a failing request that we `catch` above?

Yes, that's the thing that was happening to me.

Reply:
The message is the same inside `catch`, so it seems we can just remove this block of code? I have to check if there is a prop `deleted` in the response though and why this was need in the first place? Any of you know about the `deleted`?

Reply:
It seems we [add this only if we](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L478) `force:true` and since we pass this param, the only case it wouldn't be `true` would be if the request fails, no? I think we can remove these checks altogether. What do you think?

Reply:
Here's the corresponding API logic - https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L477-L513.

Reply:
The check was introduced in #40007. I'm guessing there are cases in CI env when a request might fail without throwing an error, so we added an extra check.

Reply:
> I think we can remove these checks altogether. What do you think?

That seems to be the case, if `deleted` is present it will be `true`, any other error would make the request fail and it'll go on the `catch`. 🤔

Reply:
> The check was introduced in #40007. I'm guessing there are cases in CI env when a request might fail without throwing an error, so we added an extra check.

Makes sense, should we merge this then?

Reply:
From the original PR:
>I'm changing the test in this PR to use an afterEach instead of afterAll. 

I think that's the main change there. I'd go with removing it, but no strong opinions here. I'll leave it for @Mamaduka 😃

Reply:
We could even use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) here :)

Reply:
> I'd go with removing it, but no strong opinions here.

That works for me as well 👍

Reply:
Just pushed the change 😄 
Thanks a lot for your help!

Reply:
Thanks so much! Can you also remove the same code block in the other util as well? After that we are good to land.

Reply:
Done! :)